### PR TITLE
toolchain: [Danger] Add guard to protect against <QueryRenderer> imports 

### PR DESCRIPTION
**Pull request by damassi**
Addresses https://artsyproduct.atlassian.net/browse/WP-44

This adds a new Danger rule that guards against the use of the default `<QueryRenderer />`, and instead directs devs to use our `<SystemQueryRenderer />`. 

Why? 

The regular `<QueryRenderer />`, which should only fire on the client, will double fetch during our server-side render pass since the network layer is shared.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error RangeError
```
/app/scripts/validateSchemas.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: Maximum call stack size exceeded
RangeError: /app/scripts/validateSchemas.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: /app/relay.config.js: Maximum call stack size exceeded
    at loadPrivatePartialConfig (/app/node_modules/@babel/core/lib/config/partial.js:85:62)
    at loadPrivatePartialConfig.next (<anonymous>)
    at loadFullConfig (/app/node_modules/@babel/core/lib/config/full.js:53:46)
    at loadFullConfig.next (<anonymous>)
    at Function.<anonymous> (/app/node_modules/@babel/core/lib/config/index.js:28:43)
    at Generator.next (<anonymous>)
    at evaluateSync (/app/node_modules/gensync/index.js:251:28)
    at Function.sync (/app/node_modules/gensync/index.js:89:14)
    at Object.<anonymous> (/app/node_modules/@babel/core/lib/config/index.js:39:61)
    at Object.exports.babelify (/app/node_modules/danger/distribution/runner/runners/utils/transpiler.js:179:45)
```
### Dangerfile
```
----------------------------------------------------------------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d7907a998cb204216bc8593597c298b5571c5c14
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please use <SystemQueryRenderer /> instead of <QueryRender />. This prevents double fetching during the server-side render pass. See: - `src/v2/test.tsx`
  }
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 80742b3a8ff641f7cd63b6d3464feee837710bcc
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please use `<SystemQueryRenderer />` instead of `<QueryRender />`. This prevents double fetching during the server-side render pass. See: - `src/v2/test.tsx`
  }
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1825369ad851291846189e82b757b47eeeabe3c9
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please use `<SystemQueryRenderer />` instead of `<QueryRender />`. This prevents double fetching during the server-side render pass. See:
> `src/v2/test.tsx`
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 227d90ca536add30eef87c7c76708d9a125ae242
</p>

File: dangerfile.ts [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
-    if (isReleasePR) {
-      const deployURL = "https://github.com/artsy/metaphysics#deployment"
-      fail(
-        `You need to promote Metaphysics from staging to production. The schema used in Reaction, is further ahead than the version in Metaphyisc production. This could cause your queries to fail. You can see how to [deploy here](${deployURL}).`
-      )
-    } else {
-      warn(
-        "The staging version of Metaphysics has breaking changes, this might not affect your build, but could mean your queries don't work the way you expect. See below for breaking changes:"
-      )
-    }
```
Comment (damassi) [inline dangerfile.ts:32]: This is now called directly from a CI job

**Pull request merged**